parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java

```diff
@@ -99,6 +104,10 @@ public static Metrics footerMetrics(ParquetMetadata metadata, Stream<FieldMetric
     MessageType parquetTypeWithIds = getParquetTypeWithIds(metadata, nameMapping);
     Schema fileSchema = ParquetSchemaUtil.convertAndPrune(parquetTypeWithIds);
 
+    Map<Integer, FieldMetrics> fieldMetricsMap = Optional.ofNullable(fieldMetrics)
+        .map(stream -> stream.collect(Collectors.toMap(FieldMetrics::id, Function.identity())))
+        .orElseGet(HashMap::new);
```

Comment:
Why not `.orElse(ImmutableMap.of())`? Seems strange to create a new map if field metrics are missing. And won't `stream.collect(Collectors.toMap(...))` return an empty map if the stream is empty? Does this need to handle a null stream or can it just assume that it will be non-null?

Reply:
I was trying to be defensive in handling null stream here which is why I had this Optional wrapper and `orElse`, but you are right the current implementation wouldn't produce a null stream. I'll add a precondition check for null stream and then directly use the stream to create the map.